angularjs/angular.d.ts

```diff
@@ -1413,13 +1402,15 @@ declare module angular {
         (data: any, headersGetter: IHttpHeadersGetter, status: number): any;
     }
 
+    type HttpHeaderType = {[requestType: string]:string|((config:IRequestConfig) => string)};
+
     interface IHttpRequestConfigHeaders {
-        [requestType: string]: string|(() => string);
-        common?: string|(() => string);
-        get?: string|(() => string);
-        post?: string|(() => string);
-        put?: string|(() => string);
-        patch?: string|(() => string);
+        [requestType: string]: any;
+        common?: any;
+        get?: any;
+        post?: any;
+        put?: any;
+        patch?: any;
```

Comment:
Before I jump to premature conclusions again, do you have any additional information on this @mmiszy ?